source/keyboardHandler.py

```diff
@@ -205,10 +205,12 @@ def internal_keyDownEvent(vkCode,scanCode,extended,injected):
 			# and not if this gesture is a modifier key
 			and not isNVDAModifierKey(vkCode,extended) and not vkCode in KeyboardInputGesture.NORMAL_MODIFIER_KEYS
 			and ( # Either of
-				# We couldn't inject in-process, and its not a console window (console windows have their own specific typed character support)
+				# We couldn't inject in-process, and its not a legacy console window (console windows have their own specific typed character support)
 				(not focus.appModule.helperLocalBindingHandle and focus.windowClassName!='ConsoleWindowClass')
 				# or the focus is within a UWP app, where WM_CHAR never gets sent 
 				or focus.windowClassName.startswith('Windows.UI.Core')
+				#Or this is a UIA console window, where WM_CHAR messages are doubled
+				or focus.windowClassName == "ConsoleWindowClass" and config.conf['UIA']['consoleUIA']

```

Comment:
I don't believe this is the right way to ensure that UIA is enabled, especially not for older versions of Windows. I think I'd rather see an instance check of the object